libsweep/src/arch/win/serial_win.cc

```diff
@@ -0,0 +1,291 @@
+#include "serial.h"
+
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <cstdio>
+#include <string>
+
+#include <windows.h>
+
+namespace sweep {
+namespace serial {
+
+typedef struct error {
+  const char* what; // always literal, do not free
+} error;
+
+typedef struct device {
+  HANDLE fd;
+  OVERLAPPED ro_;
+  OVERLAPPED wo_;
+} device;
+
+// Constructor hidden from users
+static error_s error_construct(const char* what) {
+  SWEEP_ASSERT(what);
+
+  auto out = new error{what};
+  return out;
+}
+
+const char* error_message(error_s error) {
+  SWEEP_ASSERT(error);
+
+  return error->what;
+}
+
+void error_destruct(error_s error) {
+  SWEEP_ASSERT(error);
+
+  delete error;
+}
+
+static int32_t detail_get_port_number(const char* port, error_s* error) {
+  SWEEP_ASSERT(port);
+  SWEEP_ASSERT(error);
+
+  if (strlen(port) <= 3) {
+    *error = error_construct("invalid port name");
+    return -1;
+  }
+
+  char* end;
+  long parsed_int = std::strtoll(port + 3, &end, 10);
```

Comment:
shouldn't that line be

    long parsed_int = std::strtol(port + 3, nullptr, 10);